pkg/local_object_storage/engine/delete.go

```diff
@@ -61,21 +62,29 @@ func (e *StorageEngine) delete(prm DeletePrm) (DeleteRes, error) {
 		is  bool
 		err apistatus.ObjectLocked
 	}
+	var splitInfo *objectSDK.SplitInfo
 
+	// Removal of a big object is done in multiple stages:
+	// 1. Remove the parent object. If it is locked or already removed, return immediately.
+	// 2. Otherwise, search for all objects with a particular SplitID and delete them too.
 	e.iterateOverSortedShards(prm.addr, func(_ int, sh hashedShard) (stop bool) {
 		var existsPrm shard.ExistsPrm
 		existsPrm.SetAddress(prm.addr)
 
 		resExists, err := sh.Exists(existsPrm)
 		if err != nil {
-			_, ok := err.(*objectSDK.SplitInfoError)
-			if ok || shard.IsErrRemoved(err) || shard.IsErrObjectExpired(err) {
+			if shard.IsErrRemoved(err) || shard.IsErrObjectExpired(err) {
 				return true
 			}
-			if !shard.IsErrNotFound(err) {
-				e.reportShardError(sh, "could not check object existence", err)
+
+			splitErr, ok := err.(*objectSDK.SplitInfoError)
```

Comment:
Why not `errors.As`?

Reply:
Mostly because it is faster and was already done like this

Reply:
Oh yeah, missed that it remains unchanged.